port/cpl_aws.cpp

```diff
@@ -730,6 +730,132 @@ static bool IsMachinePotentiallyEC2Instance()
 #endif
 }
 
+
+/************************************************************************/
+/*                   ReadAWSWebIdentityTokenFile()                      */
+/************************************************************************/
+
+static bool ReadAWSWebIdentityTokenFile(const std::string& osWebIdentityTokenFile,
+                                        CPLString& webIdentityToken)
+{
+    GByte *pabyOut = nullptr;
+    if( !VSIIngestFile( nullptr, osWebIdentityTokenFile.c_str(), &pabyOut, nullptr, -1 ) )
+        return false;
+
+    webIdentityToken = reinterpret_cast<char *>(pabyOut);
+
+    return !webIdentityToken.empty();
+}
+
+/************************************************************************/
+/*          GetConfigurationFromAssumeRoleWithWebIdentity()             */
+/************************************************************************/
+
+bool VSIS3HandleHelper::GetConfigurationFromAssumeRoleWithWebIdentity(bool bForceRefresh,
+                                                                      const std::string& osPathForOption,
+                                                                      CPLString& osSecretAccessKey,
+                                                                      CPLString& osAccessKeyId,
+                                                                      CPLString& osSessionToken)
+{
+    CPLMutexHolder oHolder( &ghMutex );
+    if( !bForceRefresh )
+    {
+        time_t nCurTime;
+        time(&nCurTime);
+        // Try to reuse credentials if they are still valid, but
+        // keep one minute of margin...
+        if( !gosGlobalAccessKeyId.empty() && nCurTime < gnGlobalExpiration - 60 )
+        {
+            osAccessKeyId = gosGlobalAccessKeyId;
+            osSecretAccessKey = gosGlobalSecretAccessKey;
+            osSessionToken = gosGlobalSessionToken;
+            return true;
+        }
+    }
+
+    const CPLString roleArn = VSIGetCredential(osPathForOption.c_str(), "AWS_ROLE_ARN", "");
+        if( roleArn.empty() )
+        {
+            VSIError(VSIE_AWSInvalidCredentials, "AWS_ROLE_ARN configuration option not defined");
```

Comment:
when trying different authentication methods, we can get here. Emitting an error isn't appropriate in that situation, since a later authentication method might work

Reply:
indeed moving it to CPLDebug("AWS", ...